cli/azd/internal/telemetry/telemetry_context.go

```diff
@@ -0,0 +1,50 @@
+package telemetry
+
+import (
+	"context"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/internal/telemetry/baggage"
+	"github.com/azure/azure-dev/cli/azd/internal/telemetry/fields"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func getEnvironmentAttributes(env *environment.Environment) []attribute.KeyValue {
+	return []attribute.KeyValue{
+		fields.ObjectIdKey.String(env.GetPrincipalId()),
```

Comment:
Long term I think we will want to remove the concept of "principal id" from the environment, since it's a concept tied to the current user of `azd` not the environment.  I suspect in that world we'd just end up decorating the context with this information elsewhere?

Reply:
That would be correct. Although, I suspect in the future we want to move towards storing a user profile JSON in `.azd`. We already have `azd login`, and as we move towards removing AZ CLI, we would likely have that context stored and initialized pretty early.

Reply:
> We already have azd login, and as we move towards removing AZ CLI, we would likely have that context stored and initialized pretty early.

This matches my intuition as well.